pkg/operator/cockroachdb/controller.go

```diff
@@ -187,8 +187,11 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 }
 
 func (c *ClusterController) onDelete(obj interface{}) {
-	cluster := obj.(*cockroachdbv1alpha1.Cluster).DeepCopy()
-	logger.Infof("cluster %s deleted from namespace %s", cluster.Name, cluster.Namespace)
+	cluster, ok := obj.(*cockroachdbv1alpha1.Cluster)
+	if ok {
```

Comment:
One more suggestion on the pattern to follow, I should have mentioned this before... For each of these places you check for `ok`, let's check for `!ok` and then return. This way the rest of the method can remain unchanged and the depth of code is better. For example:
```
cluster, ok := obj.(*cockroachdbv1alpha1.Cluster)
if !ok {
   return
}
cluster = cluster.DeepCopy()
logger.Infof("cluster %s deleted from namespace %s", cluster.Name, cluster.Namespace)
```

Reply:
@travisn  fixed